raysort/main.py

```diff
@@ -270,10 +284,25 @@ def get_block(i: int, d: int) -> np.ndarray:
 
 
 def get_boundaries(
-    num_map_returns: int, num_merge_returns: int = -1
+    cfg: AppConfig,
+    parts: List[PartInfo],
+    num_map_returns: int,
+    num_merge_returns: int = -1,
 ) -> Tuple[List[int], List[List[int]]]:
     if num_merge_returns == -1:
```

Comment:
do something like
```
get_boundaries_fn = get_boundaries_by_sampling if cfg.use_sampling else sortlib.get_boundaries
```

Then replace the calls to `sortlib.get_boundaries`, namely L293, 306, with `get_boundaries_fn`.

Then implement `get_boundaries_by_sampling` with the same function signature as `sortlib.get_boundaries`, and run your sampling tasks then call Isaac's function for computing the bounds.